morecantile/models.py

```diff
@@ -778,13 +778,15 @@ def tiles(
             n = min(self.bbox.top, n)
 
             for z in zooms:
-                ul_tile = self.tile(
+                nw_tile = self.tile(
                     w + LL_EPSILON, n - LL_EPSILON, z
                 )  # Not in mercantile
-                lr_tile = self.tile(e - LL_EPSILON, s + LL_EPSILON, z)
+                se_tile = self.tile(e - LL_EPSILON, s + LL_EPSILON, z)
+                minx, maxx = (nw_tile.x, se_tile.x) if nw_tile.x < se_tile.x else (se_tile.x, nw_tile.x)
+                miny, maxy = (nw_tile.y, se_tile.y) if nw_tile.y < se_tile.y else (se_tile.y, nw_tile.y)
 
-                for i in range(ul_tile.x, lr_tile.x + 1):
-                    for j in range(ul_tile.y, lr_tile.y + 1):
+                for i in range(minx, maxx + 1):
+                    for j in range(miny, maxy + 1):
```

Comment:
FYI: This originally comes from `mercantile` https://github.com/mapbox/mercantile/blob/main/mercantile/__init__.py#L541-L547 

Could you add a tests for this? (like something that was failing with previous version 🙏 )

Reply:
Got it! I guess it makes sense for `mercantile` to only handle the standard axis ordering.

Test added.